### PR TITLE
Put in a warning, and a nrow check on loading ACE data

### PR DESCRIPTION
**Pull request by arjunpur**
I noticed a bug where in modules where there were only practice trials we end up erasing the entire data frame because of this [filter](https://github.com/joaquinanguera/aceR/blob/master/R/load-csv.R#L307). The result is that when we attempt any future operations on the empty dataframe, the `load_ace_bulk` function will error out.

This PR adds a warning and some `nrow` checks to verify that the tibble we're operating on has non-zero rows.

### CODE REVIEW

**Review by monicathieu — CHANGES_REQUESTED** [review]
These changes look good overall! Thank you for catching that edge case.

I have a couple requests and one small suggested code change:

1. Code change: around line 175 of load-ace.R, consider directly returning `dat` inside that `if (nrow(dat) == 0)` statement to make explicit that no additional processing is called on `dat` if it is empty
2. Request part 1: Can you upload an example of an all-practice failing raw data file to inst/extdata/nexus? (assuming it is Nexus data, you can make a new folder at that path and add it there) You can then access it in tests using `file.path(aceR_sample_data_path("nexus"), "file-name.csv")`.
3. Request part 2: In tests/testthat/test-load-csv.R, can you write a test in testthat syntax to expect that calling `load_ace_file()` on the all-practice data file will return a 0-rows dataframe, and throws the expected warning?
4. Request part 3: Can you write a test in tests/testthat/test-ace-pipeline.R (you can put it in where line 26 is now) that tests that calling `load_ace_bulk()` on the all-practice data file returns an appropriate aceR dataframe but with the data column empty?

Let me know once those tests are written and passing, or if you have other questions! (The example tests should guide you but mind you I wrote them a long while ago and the styling may be a little hacky, so don't follow them to a T.)

**Comment by arjunpur** [comment]
That sounds good - I can make those changes! Thanks @monicathieu. Do you know if Travis CI is set up on this repository? I tried accessing the link in CONTRIBUTING.md (+ changing the repo name) but I don't think anything runs.

**Comment by arjunpur** [comment]
Another question @monicathieu : It looks like you have some unmerged changes on the `development` branch (which is where you created `inst/extdata/nexus`) - should we merge those into `master` before I merge this PR in? or should we let them be?

**Comment by monicathieu** [comment]
To your questions:

1. Once upon a LONG time ago I think Travis CI was set up to run, but no longer... I need to put that in an enhancement milestone. So many quality-of-life improvements, so little time! For the time being, if you can run the testthat tests locally and confirm they pass, that'll do.
2. Let's leave the development branch unmerged for now, as it's got a bunch of other Nexus updates that (while mostly ready to go) are not 100% tested through yet. We can merge the contents of inst/extdata/nexus across branches later!

**Comment by rrflynt** [comment]
Hi Monica,

Can you please take me off of this email chain?
Thanks

Reta Riedinger
(Flynt)

(208) 309-1286

**Comment by monicathieu** [comment]
Hi Reta @rrflynt ,

GitHub doesn't allow me to remove you from the repository, but you can stop receiving notifications about repository updates by going to the repository page https://github.com/joaquinanguera/aceR/ , clicking "Unwatch" in the top right-hand corner, and then choosing the notification setting you prefer.

**Comment by arjunpur** [comment]
Hi @monicathieu - I've addressed your comments [here](https://github.com/joaquinanguera/aceR/pull/52/commits/346cb2f77c15b69ea3197432c9a866eb47062118). Let me know what you think!

File: tests/testthat/test-ace-pipeline.R
Comment (monicathieu) [inline tests/testthat/test-ace-pipeline.R]: This test looks good! If I'm reading the diff correctly I think the `all_practice_file_path` assignment on line 29 can be deleted because it doesn't appear to be used later in this test script.

File: tests/testthat/test-load-csv.R
Comment (monicathieu) [inline tests/testthat/test-load-csv.R]: Thanks for writing this test as well! Per the current [testthat docs](https://testthat.r-lib.org/reference/expect_error.html), it looks like `capture_warnings()` is superseded, and they recommend this syntax for testing the warning and the returned object:

```suggestion
  expect_warning(
    result <- load_ace_file(all_practice_file_path, app_type = "nexus"),
    "processing resulted in an empty data frame"
  )
```

Then current line 24 that checks to make sure the result has 0 rows should still run as usual, because `result` will be assigned in the process of running `expect_warning()`.

**Review by monicathieu — COMMENTED** [review]
Reviewed changes look good overall, thank you for working on this! Will approve merging if you think my small line edits make sense (see inline comments).

File: tests/testthat/test-ace-pipeline.R
Comment (monicathieu) [inline tests/testthat/test-ace-pipeline.R]: Forgot to add the suggestion block to allow inline committing directly from the review:
```suggestion
```

**Comment by arjunpur** [comment]
Awesome thanks for the suggestions - committed them. Verified tests pass locally too

**Comment by monicathieu** [comment]
Hooray! Confirming merge ok!

**Review by monicathieu — APPROVED** [review]
Accepting!

**Pull request merged**
